midealocal/device.py

```diff
@@ -510,10 +512,9 @@ def _check_heartbeat(self, now: float) -> None:
     def run(self) -> None:
         """Run loop."""
         while self._is_run:
-            while self._socket is None:
-                if self.connect(refresh_status=True) is False:
-                    self.close_socket()
-                    time.sleep(5)
+            if not self._socket or not self.connect():
```

Comment:
This looks a little bit off. It'll call connect if socket is not none and a new socket will be open.
I didn't put my thought in this but this should be something like this:
```suggestion
            if self._socket is None and not self.connect():
```

Reply:
`if not self._socket` gives the same result as `if self._socket is None`

Reply:
Anyway, it should use ` and` instead of `or`

Reply:
not convinced at all, but I'll change it if you think it should be changed.

Reply:
if socket is not none it'll go to connect that will replace the socket, leaving an open socket.

Reply:
With suggested change, mypy complains:

![image](https://github.com/user-attachments/assets/a53c8d5c-8e38-4047-8b35-15bc9092825f)

Reply:
> if socket is not none it'll go to connect that will replace the socket, leaving an open socket.

For this we can just add a : 

```python
finally:
                self._socket.close()
```

in the connect() try block

Reply:
You didn't get it.

If you use `or`:
| socket is None | connect return | result |
|--------|--------|--------|
| True | Won't be run | won't raise socketexception |
| False | False | There is an active socket and another one is instantiated inside connect function, then socketexception is raised |
| False | True | There is an active socket and another one is instantiated inside connect function, no socketexception is raised |

Reply:
Reworked to improve socket management, even if I think it's still not optimal.
Enough to proceed with this PR.